src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/FindOneAndUpdateCommand.java

```diff
@@ -17,15 +18,26 @@
 @JsonTypeName("findOneAndUpdate")
 public record FindOneAndUpdateCommand(
     @Valid @JsonProperty("filter") FilterClause filterClause,
-    @Valid @JsonProperty("update") UpdateClause updateClause,
+    @NotNull @Valid @JsonProperty("update") UpdateClause updateClause,
     @Valid @Nullable Options options)
     implements ReadCommand, Filterable {
+
+  @Schema(
+      name = "FindOneAndUpdateCommand.Options",
+      description = "Options for `findOneAndUpdate` command.")
   public record Options(
-      @Valid
-          @Nullable
+      @Nullable
           @Pattern(
               regexp = "(after|before)",
               message = "returnDocument value can only be 'before' or 'after'")
+          @Schema(
+              description =
+                  "Specifies which document to perform the projection on. If `before` the projection is performed on the document before the update is applied, if `after` the document projection is from the document after the update.",
+              defaultValue = "before")
           String returnDocument,
-      boolean upsert) {}
+      @Schema(
+              description =
+                  "When `true`, if no documents match the `filter` clause the command will create a new _empty_ document and apply the `update` clause to the empty document.",
```

Comment:
Strictly speaking it's not exactly empty I think, as some values are initialized from filter clause.

@maheshrajamani Is this correct? And if so, how are default values from filter clause selected/applied (only equality ones)

Reply:
Yes, When new document is created all equals filter's are added to the document.

Reply:
Will fix. Should be fixed in update commands as well

Reply:
@maheshrajamani is right about this one, any equals filters are added to the document before updates are applied.